packages/backend-storage/src/storage_access_policy_factory.ts

```diff
@@ -63,9 +63,15 @@ export class StorageAccessPolicyFactory {
       });
     }
 
-    return new Policy(this.stack, `${this.namePrefix}${this.policyCount++}`, {
```

Comment:
The previous name always started from 1,2,3 for each bucket. We need to give it different names.

I can try if the `num` in the construct.ts is available in a new commit.